lld/ELF/LinkerScript.cpp

```diff
@@ -1230,6 +1230,9 @@ bool LinkerScript::assignOffsets(OutputSection *sec) {
   if (sec->firstInOverlay)
     state->overlaySize = 0;
 
+  bool synthesizeAlign = ctx.arg.relocatable && ctx.arg.relax &&
+                         (sec->flags & SHF_EXECINSTR) &&
+                         ctx.arg.emachine == EM_RISCV;
```

Comment:
It might be unlikely that another target would need this, but conceptually, it seems like the determination of whether a call to `maybeSynthesizeAlign` is needed should be a target hook.